src/StrMacros.php

```diff
@@ -14,10 +14,24 @@ public function between()
             if (is_null($end)) {
                 $end = $beginning;
             }
+
             return Str::before(Str::after($subject, $beginning), $end);
         };
     }
 
+    public function extract()
+    {
+        return function ($string, $pattern) {
+            if (@preg_match($pattern, $string) === false) {
```

Comment:
It looks like the `preg_match` will return integer value.

And I think this will not return the bool value.

Reply:
From [php.net](https://www.php.net/manual/en/function.preg-match.php#refsect1-function.preg-match-returnvalues):

"**preg_match()** returns 1 if the **pattern** matches given **subject**, 0 if it does not, or **FALSE** if an error occurred.

Reply:
That makes sense.